### PR TITLE
Dont let validation issues prevent saving

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Saving content 'saves the content' no matter the validity of the data. This is how we want it to be, so Content Editors are able to save work in progress.
But if any validation message is shown in the UI, this prevents a save from begin carried out. Making the experience inconsistent, this PR changes that and makes the save feature carry out its action no matter the validation state.

And then the UI should not be red, as the messages are not alerting, just a warning. Therefor they are also turned into warning-style(yellow)

image

I also did a little recording, describing the old issue and showing the solution:
https://user-images.githubusercontent.com/6791648/115400305-66dacf80-a1e9-11eb-98ad-c34241b90862.mov

**Test Notes**
- Test for variant and invariant content as code is different for these scenarios.
- Make sure to test nested validation & complex editors.
- test that we didn't touch other scenarios, like publish etc. (Publish must still be prevented by validation issues)

fixes https://github.com/umbraco/Umbraco-CMS/issues/7484

---
_This item has been added to our backlog [AB#10080](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/10080)_

### CODE REVIEW

**Comment by warrenbuckley** [comment]
@nielslyngsoe can you fix up the merge conflict whilst I review and test please

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js [hunk 1]
```diff
@@ -468,15 +469,19 @@
...
+                }
                 ensureDirtyIsSetIfAnyVariantIsDirty();
 
                 return $q.when(data);
             },
                 function (err) {
                     syncTreeNode($scope.content, $scope.content.path);
 
-                    resetNestedFieldValiation(fieldsToRollback);
+                    if($scope.contentForm.$invalid !== true) {
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js:482]: Ah so much inversion - not an easy line to read, so basically the form is valid then?! 🤯

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js [hunk 2]
```diff
@@ -729,9 +734,9 @@
             clearNotifications($scope.content);
             // TODO: Add "..." to save button label if there are more than one variant to publish - currently it just adds the elipses if there's more than 1 variant
             if (hasVariants($scope.content)) {
-                //before we launch the dialog we want to execute all client side validations first
-                if (formHelper.submitForm({ scope: $scope, action: "openSaveDialog" })) {
 
+                //before we launch the dialog we want to execute all client side validations first
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js:738]: Is this comment correct still, if you are skipping client validation?!?

**Review by warrenbuckley — COMMENTED** [review]
Code mostly reads fine, can you just please answer my couple tiny questions @nielslyngsoe & fix up the merge conflict

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js (hunk 1 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js:484]: Yep, if it is valid then we performe this if.

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js (hunk 2 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js:744]: Well, we still launch them, we just don't allow them to be blocking. And still this is only affected by save, not save & publish.

**Pull request merged**